library/extractor/src/main/java/com/google/android/exoplayer2/extractor/ts/DefaultTsPayloadReaderFactory.java

```diff
@@ -176,11 +182,41 @@ public TsPayloadReader createPayloadReader(int streamType, EsInfo esInfo) {
       case TsExtractor.TS_STREAM_TYPE_DVBSUBS:
         return new PesReader(
             new DvbSubtitleReader(esInfo.dvbSubtitleInfos));
+      case TsExtractor.TS_STREAM_TYPE_AIT:
+        return new SectionReader(new SectionPassthrough(MimeTypes.APPLICATION_AIT));
       default:
         return null;
     }
   }
 
+  public class SectionPassthrough implements SectionPayloadReader {
+    private TimestampAdjuster timestampAdjuster = null;
+    private final String mimeType;
+    private TrackOutput output;
```

Comment:
Can this be `final`?

Reply:
I can't see how, since output can be created only in init() function not in constructor.
Looking at other Readers (Ac3Reader), I see that trackouput is `@MonotonicNonNull` which I'll add.